plugins/redis/redis.cpp

```diff
@@ -0,0 +1,904 @@
+    returnLength = static_cast<size32_t>(returnSize/sizeof(UChar));
+}
+ECL_REDIS_API void ECL_REDIS_CALL SyncRGetUtf8(ICodeContext * ctx, size32_t & returnLength, char * & returnValue, const char * key, const char * options, unsigned __int64 database, const char * password, unsigned __int64 timeout)
+{
+    size_t returnSize;
+    SyncRGet(ctx, options, key, returnSize, returnValue, database, password, timeout);
+    returnLength = static_cast<size32_t>(rtlUtf8Length(returnSize, returnValue));
+}
+ECL_REDIS_API void ECL_REDIS_CALL SyncRGetData(ICodeContext * ctx, size32_t & returnSize, void * & returnValue, const char * key, const char * options, unsigned __int64 database, const char * password, unsigned __int64 timeout)
+{
+    size_t _returnSize;
+    SyncRGet(ctx, options, key, _returnSize, returnValue, database, password, timeout);
+    returnSize = static_cast<size32_t>(_returnSize);
+}
+//----------------------------------LOCK------------------------------------------
+//-----------------------------------SET-----------------------------------------
+//Set pointer types
+void SyncLockRSet(ICodeContext * ctx, const char * _options, const char * key, size32_t valueSize, const char * value, unsigned __int64 database, unsigned expire, const char * password, unsigned __int64 _timeout)
+{
+    Owned<Connection> master = Connection::createConnection(ctx, _options, database, password, _timeout);
+    master->lockSet(ctx, key, valueSize, value, expire);
+}
+//--INNER--
+void Connection::lockSet(ICodeContext * ctx, const char * key, size32_t valueSize, const char * value, unsigned expire)
+{
+    const char * _value = reinterpret_cast<const char *>(value);//Do this even for char * to prevent compiler complaining
+    handleLockOnSet(ctx, key, _value, (size_t)valueSize, expire);
+}
+//-------------------------------------------GET-----------------------------------------
+//--OUTER--
+void SyncLockRGet(ICodeContext * ctx, const char * options, const char * key, size_t & returnSize, char * & returnValue, unsigned __int64 database, const char * password, unsigned __int64 _timeout)
+{
+    Owned<Connection> master = Connection::createConnection(ctx, options, database, password, _timeout);
+    master->lockGet(ctx, key, returnSize, returnValue, password);
+}
+//--INNER--
+void Connection::lockGet(ICodeContext * ctx, const char * key, size_t & returnSize, char * & returnValue, const char * password)
+{
+    MemoryAttr retVal;
+    handleLockOnGet(ctx, key, &retVal, password);
+    returnSize = retVal.length();
+    returnValue = reinterpret_cast<char*>(retVal.detach());
+}
+//---------------------------------------------------------------------------------------
+void Connection::encodeChannel(StringBuffer & channel, const char * key) const
+{
+    channel.append(REDIS_LOCK_PREFIX).append("_").append(key).append("_").append(database).append("_").append(ip.str()).append("_").append(port);
```

Comment:
should the ip be included in the encoded channel?  What if there are multiple balanced redis servers on different ips (question from ignorance)